aipg/rag/service.py

```
@@ -0,0 +1,96 @@
+            logger.warning(
+                f"Empty or placeholder content detected for topic '{topic}' "
+                f"and project '{micro_project.topic}': content='{micro_project.raw_markdown[:100]}...'"
+            )
+            raise RuntimeError(
+                f"Cannot save empty or placeholder content for topic '{topic}' "
+                f"and project '{micro_project.topic}'. Content must contain meaningful project data."
+            )
+
+        embeddings = await self.embedder.embedding_processor([topic])
+        if not embeddings:
+            raise RuntimeError(f"Failed to generate embedding for topic: '{topic}'")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)